ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplExternalTables.java

```diff
@@ -162,7 +194,77 @@ private void dirLocationToCopy(String tableName, FileList fileList, Path sourceP
       targetPath = new Path(Utils.replaceHost(targetPath.toString(), sourcePath.toUri().getHost()));
       sourcePath = new Path(Utils.replaceHost(sourcePath.toString(), remoteNS));
     }
-    fileList.add(new DirCopyWork(tableName, sourcePath, targetPath).convertToString());
+    fileList.add(new DirCopyWork(tableName, sourcePath, targetPath, copyMode, snapshotPrefix).convertToString());
+  }
+
+  private SnapshotUtils.SnapshotCopyMode createSnapshotsAtSource(Path sourcePath, String snapshotPrefix,
+      boolean isSnapshotEnabled, HiveConf conf, SnapshotUtils.ReplSnapshotCount replSnapshotCount, FileList snapPathFileList,
+      ArrayList<String> prevSnaps, boolean isBootstrap) throws IOException {
+    if (!isSnapshotEnabled) {
+      LOG.info("Snapshot copy not enabled for path {} Will use normal distCp for copying data.", sourcePath);
+      return FALLBACK_COPY;
+    }
+    DistributedFileSystem sourceDfs = SnapshotUtils.getDFS(sourcePath, conf);
+    try {
+      if(isBootstrap) {
+        // Delete any pre existing snapshots.
+        SnapshotUtils.deleteSnapshotSafe(sourceDfs, sourcePath, firstSnapshot(snapshotPrefix));
```

Comment:
if deletion fails what happens?

Reply:
Changed behaviour, If it exists it will delete, and if the deletion fails, replication fails